### PR TITLE
Fail on warnings for Jellyfin.Server

### PR DESCRIPTION
**Pull request by Bond-009**
CI will now fail on style issues inside of the `Jellyfin.Server` project
Time to start on `Emby.Serer.Implementations` ;)

### CODE REVIEW

File: Emby.Server.Implementations/Activity/ActivityLogEntryPoint.cs [hunk 1]
```diff
@@ -349,18 +324,7 @@ void _userManager_UserCreated(object sender, GenericEventArgs<User> e)
             });
         }
 
-        void _subManager_SubtitlesDownloaded(object sender, SubtitleDownloadEventArgs e)
```
Comment (JustAMan) [inline Emby.Server.Implementations/Activity/ActivityLogEntryPoint.cs]: I wonder why this was unused...

File: Jellyfin.Server/Jellyfin.Server.csproj [hunk 2]
```diff
@@ -12,7 +12,8 @@
     <!-- We need C# 7.1 for async main-->
     <LangVersion>latest</LangVersion>
     <!-- Disable documentation warnings (for now) -->
-    <NoWarn>SA1600;CS1591</NoWarn>
+    <NoWarn>SA1600;SA1601;CS1591</NoWarn>
```
Comment (JustAMan) [inline Jellyfin.Server/Jellyfin.Server.csproj]: is `SA1601` also a documentation warning?

File: Jellyfin.Server/SocketSharp/WebSocketSharpRequest.cs [hunk 3]
```diff
@@ -40,10 +53,14 @@ public WebSocketSharpRequest(HttpListenerContext httpContext, string operationNa
 
         public object Dto { get; set; }
 
+#pragma warning disable CA1056
```
Comment (JustAMan) [inline Jellyfin.Server/SocketSharp/WebSocketSharpRequest.cs]: can we have a comment explaining what this warning is about?

File: Jellyfin.Server/SocketSharp/WebSocketSharpRequest.cs [hunk 4]
```diff
@@ -82,7 +98,134 @@ public string RemoteIp
             }
         }
 
-        private static readonly char[] HttpTrimCharacters = new char[] { (char)0x09, (char)0xA, (char)0xB, (char)0xC, (char)0xD, (char)0x20 };
+        public bool IsSecureConnection => request.IsSecureConnection || XForwardedProtocol == "https";
+
+#pragma warning disable CA1819
```
Comment (JustAMan) [inline Jellyfin.Server/SocketSharp/WebSocketSharpRequest.cs]: ditto

File: Jellyfin.Server/SocketSharp/WebSocketSharpRequest.cs [hunk 5]
```diff
@@ -82,7 +98,134 @@ public string RemoteIp
...
+        }
+
+        public Uri UrlReferrer => request.UrlReferrer;
+
+        public Stream InputStream => request.InputStream;
+
+        public long ContentLength => request.ContentLength64;
+
+// TODO: Fix
+#pragma warning disable CA1819
```
Comment (JustAMan) [inline Jellyfin.Server/SocketSharp/WebSocketSharpRequest.cs]: same

File: Jellyfin.Server/SocketSharp/WebSocketSharpRequest.cs [hunk 6]
```diff
@@ -82,7 +98,134 @@ public string RemoteIp
...
+
+#pragma warning disable CA1819
+
+        public string[] AcceptTypes => request.AcceptTypes;
+
+#pragma warning restore CA1819
+
+        public Dictionary<string, object> Items => items ?? (items = new Dictionary<string, object>());
+
+#pragma warning disable CA1721
```
Comment (JustAMan) [inline Jellyfin.Server/SocketSharp/WebSocketSharpRequest.cs]: same

File: Jellyfin.Server/Jellyfin.Server.csproj (hunk 2 above)
Comment (Bond-009) [reply Jellyfin.Server/Jellyfin.Server.csproj]: Yep

**Comment by JustAMan** [comment]
@Bond-009 I think I saw exactly these changes included in #1023, can you strip them out of that PR and leave them here?

File: Emby.Server.Implementations/Activity/ActivityLogEntryPoint.cs (hunk 1 above)
Comment (Bond-009) [reply Emby.Server.Implementations/Activity/ActivityLogEntryPoint.cs]: Because the event doesn't exist anymore.

**Comment by Bond-009** [comment]
@JustAMan Easiest solution is to merge this one first ;)

**Pull request merged**